src/math/CurveFitting.cc

```diff
@@ -510,28 +514,26 @@ namespace pfs { namespace drp { namespace stella { namespace math {
       else{
         i = 0;
       }
-      for (j = i; j <= I_Degree_In; j++){
-        (*P_D_A2_Covar)[ ndarray::makeVector( j, p-j ) ] = D_Sum;
+      for (j = i; j <= static_cast<int>(I_Degree_In); j++){
+        covarTemp[ ndarray::makeVector( j, p-j ) ] = D_Sum;
```

Comment:
Does I_Degree_In need to be size_t?  int would be more natural, or unpack it into a const int at the top of the function.